### ISSUE TITLE
Migrate all DAG details to existing grid view dag details panel

### ISSUE THREAD

**Issue by bbovenzi** [apache/airflow#29859]
- [x] Add any important information in `/dags/{dag_id}/details` to the dag details panel in grid view (when a user doesn't have a task or run selected).
- [ ] Redirect all urls
- [ ] Delete old dag details page

**Comment by Adaverse** [comment]
Can I take this up?

**Comment by potiuk** [comment]
Sure

**Comment by Adaverse** [comment]
Hello guys,
Was going through the data fetched in `/dags/{dag_id}/details` the one which is to be replaced by `/api/v1.airflow_api_connexion_endpoints_dag_endpoint_get_dag_details` (Please correct me if I'm wrong here)

Some data is not provided by the API endpoint, especially the ones under **DagModel debug information** present in `/dags/{dag_id}/details`. 

Just wanted to confirm if it is intentional or if it is expected to add this information to the endpoint.

Below is the missing info ->
- has_import_errors
- has_task_concurrency_limits
- is_paused_at_creation
- last_expired
- last_parsed_time
- last_pickled
- metadata
- next_dagrun
- next_dagrun_create_after
- next_dagrun_data_interval
- next_dagrun_data_interval_end
- next_dagrun_data_interval_start
- parent_dag
- processor_subdir
- registry
- root_dag_id
- safe_dag_id
- scheduler_lock
- timetable_description

**Comment by Adaverse** [comment]
Hi @bbovenzi,

There are information missing from the `/dags/{dag_id}/details` API, especially the ones in the **DagModel debug information** in current (old) details. Shall we add this info to the API or shall this be taken up separately?

**Comment by bbovenzi** [comment]
Yeah, let's go forward with what we have. We can add those in later.

**Comment by Adaverse** [comment]
Is it fine to modify this  [this](https://github.com/apache/airflow/blob/54ac9234a8154325b13202e558c5fd8798319c7a/airflow/www/views.py#L1289) function in `views.py` to return json the way UI expects @bbovenzi?

**Comment by bbovenzi** [comment]
> Is it fine to modify this [this](https://github.com/apache/airflow/blob/54ac9234a8154325b13202e558c5fd8798319c7a/airflow/www/views.py#L1289) function in `views.py` to return json the way UI expects @bbovenzi?

Sure! I think theres a few places where we pass `response-type: application/json` for that exact reason. Are you trying to get info that the REST API does not have?

**Comment by Adaverse** [comment]
Yes! Still in double mind. Have already raised issue on it. So in future we will get all the data through api itself. So I think best way would be to first show all the data that's coming from `dag/dag_id` and have a button which on pressing will show all data that's there in the details endpoint but not in the dags endpoint. Hence even if in future the details returns more data that will be incorporated automatically without any tweaks in the UI.
